magenta-lib/src/main/scala/magenta/deployment_type/Lambda.scala

```diff
@@ -1,11 +1,14 @@
 package magenta.deployment_type
 
 import magenta.artifact.S3Path
-import magenta.tasks.{S3Upload, UpdateS3Lambda}
+import magenta.tasks.{S3Upload, SSM, STS, UpdateS3Lambda, S3 => S3Tasks}
 import magenta.{DeployParameters, DeployReporter, DeployTarget, DeploymentPackage, KeyRing, Region, Stack}
 import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.ssm.SsmClient
 
-object Lambda extends DeploymentType  {
+object Lambda extends Lambda
+
+trait Lambda extends DeploymentType with BucketParameters {
```

Comment:
just interested why the object now is a trait? is it for testability or something else?

Reply:
Mainly to allow this SSM lookup behaviour to be shared by any task that uses S3. Ideally the autoscaling and s3upload tasks should also be able to use this behaviour.

Reply:
fair enough, so I take it the way things are designed, inheritance is the only way to reuse it.  I generally avoid inheritance as it can get out of hand, but it's ok when used judiciously as it cleans up the syntax https://www.thoughtworks.com/insights/blog/composition-vs-inheritance-how-choose